docs/console/incremental-guide.md

```diff
@@ -0,0 +1,571 @@
+
+```
+/NewTypes
+|__/Types
+|__Program.cs
+|__project.json
+```
+
+Now add some new types to the folder:
+
+```
+/NewTypes
+|__/Types
+   |__Type1.cs
+   |__Type2.cs
+|__Program.cs
+|__project.json
+```
+
+Now give them all the same namespace so you can include them in your `Program.cs`.
+
+### Example: Pet Types
+
+This example creates two new types, `Dog` and `Cat`, and has them implement an interface, `IPet`.
+
+Folder Structure:
+
+```
+/NewTypes
+|__/Types
+   |__Dog.cs
+   |__Cat.cs
+   |__IPet.cs
+|__Program.cs
+|__project.json
+```
+
+`IPet.cs`:
+```csharp
+using System;
+
+namespace Pets
+{
+	public interface IPet
+	{
+		string TalkToOwner();
+	}
+}
+```
+
+`Dog.cs`:
+```csharp
+using System;
+
+namespace Pets
+{
+	public class Dog : IPet
+	{
+		public string TalkToOwner() => "Woof!";
+	}
+}
+```
+
+`Cat.cs`:
+```csharp
+using System;
+
+namespace Pets
+{
+	public class Cat : IPet
+	{
+		public string TalkToOwner() => "Meow!";
+	}
+}
+```
+
+`Program.cs`:
+```csharp
+using System;
+using Pets;
+using System.Collections.Generic;
+
+namespace ConsoleApplication
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            List<IPet> pets = new List<IPet>
+            {
+                new Dog(),
+                new Cat()  
+            };
+            
+            foreach (var pet in pets)
+            {
+                Console.WriteLine(pet.TalkToOwner());
+            }
+        }
+    }
+}
+```
+
+`project.json`:
+```javascript
+{
+    "version": "1.0.0-*",
+    "compilationOptions": {
+        "emitEntryPoint": true
+    },
+
+    "dependencies": {
+        "Microsoft.NETCore.Runtime": "1.0.1-beta-*",
+        "System.IO": "4.0.11-beta-*",
+        "System.Console": "4.0.0-beta-*",
+        "System.Runtime": "4.0.21-beta-*",
+        "System.Collections": "4.0.11-beta-23516"
+    },
+
+    "frameworks": {
+        "dnxcore50": { }
+    }
+}
+```
+
+And if you run this:
+
+```
+$ dotnet restore
+$ dotnet run
+Woof!
+Meow!
+```
+
+New pet types can be added (such as a `Bird`), extending this project.
+
+## Testing your Console App
+
+You'll probably be wanting to test your projects at some point.  Here's a good way to do it:
+
+1. Move any source of your existing project into a new `src` folder.
+
+   ```
+   /Project
+   |__/src
+   ```
+
+2. Create a `/test` directory.
+
+   ```
+   /Project
+   |__/src
+   |__/test
+   ```
+
+3. Create a new `global.json` file:
+
+   ```
+   /Project
+   |__/src
+   |__/test
+   |__global.json
+   ```
+
+   `global.json`:
+   ```javascript
+   {
+   	"projects": [
+   		"src", "test"
+   	]
+   }
+   ```
+
+   This file tells the `dotnet` driver that this is a multi-project system, which allows it to look for dependencies in more than just the current folder it happens to be executing in.  This is important because it allows you to place a dependency on the code under test in your test project.
```

Comment:
We've been using the term 'driver' ourselves. We should decide if that's the term we want to project.